packages/@lwc/ssr-compiler/src/compile-js/index.ts

```diff
@@ -47,16 +47,23 @@ const visitors: Visitors = {
     },
     ClassDeclaration(path, state) {
         const { node } = path;
-        if (!node?.superClass) {
-            return;
-        }
-        // Assume everything with a superclass is an LWC component
-        state.isLWC = true;
-        if (node.id) {
-            state.lwcClassName = node.id.name;
-        } else {
-            node.id = b.identifier('DefaultComponentName');
-            state.lwcClassName = 'DefaultComponentName';
+        if (
+            node?.superClass &&
+            // export default class extends LightningElement {}
+            (is.exportDefaultDeclaration(path.parentPath) ||
+                // class Cmp extends LightningElement {}; export default Cmp
+                path.scope
+                    ?.getBinding(node.id.name)
+                    ?.references.some((ref) => is.exportDefaultDeclaration(ref.parent)))
+        ) {
+            // If it's a default-exported class with a superclass, then it's an LWC component!
+            state.isLWC = true;
+            if (node.id) {
+                state.lwcClassName = node.id.name;
+            } else {
+                node.id = b.identifier('DefaultComponentName');
+                state.lwcClassName = 'DefaultComponentName';
+            }
```

Comment:
There is actually a lot of logic that the babel-plugin-component currently does to try to distinguish `LightningElement` classes from non-LE classes:

https://github.com/salesforce/lwc/blob/fb163f7e3c555f093f9c17b4f70446a96ee11662/packages/%40lwc/babel-plugin-component/src/component.ts#L38-L45

https://github.com/salesforce/lwc/blob/fb163f7e3c555f093f9c17b4f70446a96ee11662/packages/%40lwc/babel-plugin-component/src/decorators/index.ts#L289-L295

The first controls `registerComponent` – if a component is not registered, then we'll throw an exception at runtime.

The second controls whether decorators are transformed.

I guess at the end of the day we only have to be 1-to-1 compatible if 1) a compiler error is not going to be thrown somewhere, and 2) the `ssr-compiler` is stricter than the pre-existing toolchain. (#<span></span>1 is because, on core, components will be compiled both for SSR and CSR, so if CSR throws then we are fine.)

So I  guess I am saying this is fine, but we will probably keep discovering edge cases as we go.